File: app/src/main/java/endroad/birusa/ViewHolders.java
package endroad.birusa;


import android.content.Context;
import android.view.View;
import android.widget.TextView;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;
import endroad.birusa.model.Message;
import ru.endroad.birusa.R;
import ru.endroad.birusa.component.session.model.User;

public class ViewHolders {

    public static class MessageHolder extends RecyclerView.ViewHolder {

        private final TextView mTextField;
        private final TextView mDateField;
        private final TextView mDateFieldMyself;
        private final TextView mAuthorFieldText;

        private String author;
        private final CardView mCardView;

        public String getAuthor() {
            return author;
            //return  mAuthorFieldText.getText();
        }


        public MessageHolder(View itemView) {
            super(itemView);
            mTextField = (TextView) itemView.findViewById(R.id.view_text);
            mDateField = (TextView) itemView.findViewById(R.id.view_date);
            mDateFieldMyself = (TextView) itemView.findViewById(R.id.view_date_myself);
            mCardView = (CardView) itemView.findViewById(R.id.view_card);
            mAuthorFieldText = (TextView) itemView.findViewById(R.id.view_author_text);

        }

        public void bindData(Context context, Message message, View.OnClickListener onClickListener) {

            if (FirebaseAuth.getInstance().getCurrentUser().getUid().compareTo(message.getUid()) == 0) {

                mDateField.setVisibility(View.GONE);
                mDateFieldMyself.setVisibility(View.VISIBLE);
                mDateFieldMyself.setText(message.getDateText());

                setAvatar(message.getUid());

            } else {

                mDateFieldMyself.setVisibility(View.GONE);
                mDateField.setVisibility(View.VISIBLE);
                mDateField.setText(message.getDateText());

                setAvatar(message.getUid());
                mCardView.setOnClickListener(onClickListener);
            }

            mTextField.setText(message.getText());

        }


        void setAvatar(String userId) {
            FirebaseDatabase.getInstance().getReference().child("users").child(userId).addListenerForSingleValueEvent(
                    new ValueEventListener() {
                        @Override
                        public void onDataChange(DataSnapshot dataSnapshot) {
                            // Get user value
                            User user = dataSnapshot.getValue(User.class);
                            if (user == null) {
                                return;
                            }
                            mAuthorFieldText.setText(user.getNick());
                            author = user.getNick();
                        }

                        @Override
                        public void onCancelled(DatabaseError databaseError) {
                        }
                    });
        }

    }
}

File: feature.chat/src/main/java/ru/endroad/birusa/feature/chat/view/ViewHolders.kt
package ru.endroad.birusa.feature.chat.view

import android.view.View
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener
import ru.endroad.birusa.component.session.model.User
import ru.endroad.birusa.feature.chat.R
import ru.endroad.birusa.feature.chat.model.Message

class ViewHolders {
	class MessageHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
		private val mTextField = itemView.findViewById<TextView>(R.id.view_text)
		private val mDateField = itemView.findViewById<TextView>(R.id.view_date)
		private val mDateFieldMyself = itemView.findViewById<TextView>(R.id.view_date_myself)
		private val mAuthorFieldText = itemView.findViewById<TextView>(R.id.view_author_text)
		//return  mAuthorFieldText.getText();
		var author: String? = null
			private set
		private val mCardView = itemView.findViewById<CardView>(R.id.view_card)

		fun bindData(message: Message, onClickListener: View.OnClickListener?) {
			if (FirebaseAuth.getInstance().currentUser?.uid == message.uid) {
				mDateField.visibility = View.GONE
				mDateFieldMyself.visibility = View.VISIBLE
				mDateFieldMyself.text = message.dateText
				setAvatar(message.uid)
			} else {
				mDateFieldMyself.visibility = View.GONE
				mDateField.visibility = View.VISIBLE
				mDateField.text = message.dateText
				setAvatar(message.uid)
				mCardView.setOnClickListener(onClickListener)
			}
			mTextField.text = message.text
		}

		private fun setAvatar(userId: String?) {
			FirebaseDatabase.getInstance().reference.child("users").child(userId ?: "").addListenerForSingleValueEvent(
				object : ValueEventListener {
					override fun onDataChange(dataSnapshot: DataSnapshot) { // Get user value
						val user = dataSnapshot.getValue(User::class.java) ?: return
						mAuthorFieldText.text = user.nick
						author = user.nick
					}

					override fun onCancelled(databaseError: DatabaseError) {}
				})
		}

	}
}